q2_demux/_filter.py

```diff
@@ -0,0 +1,42 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2019, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import pandas as pd
+
+from qiime2 import Metadata
+from qiime2.util import duplicate
+from q2_demux._summarize import _PlotQualView
+from q2_types.per_sample_sequences import \
+    CasavaOneEightSingleLanePerSampleDirFmt
+
+
+def filter_samples(samples: _PlotQualView, metadata: Metadata,
+                   where: str = None, exclude_ids: bool = False) \
+                   -> CasavaOneEightSingleLanePerSampleDirFmt:
+    dir = CasavaOneEightSingleLanePerSampleDirFmt()
```

Comment:
`dir` is a builtin (though one of the less useful ones), good alternative names here could be: `dirfmt` or `dir_fmt`

Reply:
actually `output` would be even better (or `results`)